### PR TITLE
Roger Studio: TIPSv2 accuracy fixes + ADE20K bench (25.87 mIoU, beats published)

### PR DESCRIPTION
**Pull request by 2imi9**
## Summary

Three real bugs fixed in the TIPSv2 pipeline (verified live + against an official benchmark) plus a reproducible ADE20K mIoU bench script that puts every change behind a number. After the fixes + two literature-derived improvements (TCL prompt-template averaging + SegEarth-OV M-SA on the last block), TIPSv2-L/14 lands at **25.87 mIoU on full ADE20K val (n=2000) — +0.81 above the published 25.06 reference**.

### Bugs

1. **Frontend never seeded `datasets` from the backend on mount** ([App.tsx:454-463](frontend/src/App.tsx)).
   `useState<DatasetInfo[]>([])` was only mutated by upload / auto-label / delete handlers — there was no initial fetch. Result: a cold page load showed "Upload a GeoTIFF first" in TIPSv2Panel even though `/api/datasets` was returning 13 GeoTIFFs persisted from prior sessions. Added a `useEffect → listDatasets()` on mount.

2. **Hardcoded softmax scale produced near-uniform softmax** ([tipsv2_labeler.py:113-129](backend/app/services/tipsv2_labeler.py)).
   Code used `F.softmax(sim * 10, ...)` — but the model's trained `temperature` is ~0.005 (B/14) so the real logit scale is `1/0.005 ≈ 200`. With `*10`, 8-class avg confidence sat at 0.15 (random=0.125). After fix: 0.74. ADE20K-L/14 mIoU went from random baseline (~0.67%) to 17 % alone with this fix.

3. **Polygon `area_m2` was computed in degrees² after WGS84 reprojection** ([tipsv2_labeler.py:432-450](backend/app/services/tipsv2_labeler.py)).
   `poly.area` was called *after* reprojecting to EPSG:4326, returning degrees². For sub-km features at 35°N, that's ~1e-10 of the real m² → every polygon's reported area collapsed to 0, breaking the percentage roll-up. Fixed with geodesic area via `pyproj.Geod`.

### Sidebar regression fix

4. **Sidebar collapse stuck mid-animation** ([Sidebar.tsx:317](frontend/src/components/Sidebar.tsx)).
   `transition-[width]` was preventing the width Tailwind class from flipping in the browser. Removed the transition; collapse now flips cleanly at the boundary. Caught during the manual UI walk earlier in the session.

### New: TIPSv2 dense-encode helpers + ablation knobs

In [tipsv2_labeler.py](backend/app/services/tipsv2_labeler.py):

- `PROMPT_TEMPLATES` — the canonical 9 TCL templates from the official TIPSv2 release (`pytorch/TIPS_zeroshot_segmentation.ipynb`, `_TCL_PROMPTS`).
- `encode_text_with_templates(model, class_terms)` — averages L2-normalised text embeddings across all 9 templates.
- `_encode_image_dense(model, pixel_values, *, attn_mode, cls_subtract, drop_residual)` — dense readout from the encoder with selectable last-block modifications:
  - `attn_mode="default"` — unchanged
  - `attn_mode="values"` — MaskCLIP / TIPSv2 official `encode_image_value_attention`: `ls1(proj(v))`, skip MLP residual
  - `attn_mode="msa"` — SegEarth-OV (CVPR'25) modulated self-attention: sum of q-q, k-k, v-v softmax-attentions applied to V
  - `cls_subtract=λ` — SegEarth-OV Eq. 9 CLS-bias subtraction
  - `drop_residual=True` — ClearCLIP (ECCV'24) modification #1: skip the attention residual entirely

Existing `classify_image_zeroshot` and `auto_label_geotiff_tipsv2` paths are untouched — these are additive helpers used by the bench. Production paths default to the unchanged behaviour.

### New: ADE20K mIoU bench

[backend/scripts/bench_tipsv2_ade20k.py](backend/scripts/bench_tipsv2_ade20k.py) — auto-extracts the canonical MIT release zip into `backend/data/ade20k/`, runs sliding-window TIPSv2 across the val split, and reports per-class IoU + mean IoU + pixel accuracy. CLI flags map to all the dense-encode knobs above. Rerun any time with:

```
python scripts/bench_tipsv2_ade20k.py --model l14 --templates --attn-mode msa
```

## Results — ADE20K val (TIPSv2-L/14, RTX 5090, BILINEAR upsample, stride 336)

| Variant | n | mIoU | pixel acc |
| --- | ---:| ---:| ---:|
| baseline (single template, raw patch tokens) | 200 | 17.05 % | 30.49 % |
| + 9-TCL templates only | 200 | 18.68 % | 35.46 % |
| + values trick only | 200 | 22.01 % | 45.25 % |
| + 9-TCL + values | 200 | 23.12 % | 49.59 % |
| + 9-TCL + values + λ=0.3 CLS-subtract | 200 | 21.09 % | 44.54 % |
| + 9-TCL + values + drop residual | 200 | 22.77 % | 52.20 % |
| + 9-TCL + M-SA | 200 | 23.62 % | 51.73 % |
| + 9-TCL + M-SA + λ=0.3 CLS-subtract | 200 | 23.58 % | 51.27 % |
| **+ 9-TCL + values — full** | **2000** | **24.99 %** | 49.75 % |
| **+ 9-TCL + M-SA — full** | **2000** | **25.87 %** ⭐ | **52.96 %** |
| _published TIPSv2-L/14 reference (values + 9 templates)_ | _2000_ | _25.06 %_ | _—_ |

**Headline:** the values+templates run reproduces the published 25.06 within 0.07 (24.99). Switching to M-SA pushes us to 25.87 — beating the published number by +0.81. Everything verified end-to-end.

**Surprise:** SegEarth-OV's CLS-bias subtraction (λ=0.3) HURT TIPSv2 (23.12 → 21.09 with values; 23.62 → 23.58 with M-SA). The hypothesis is that TIPSv2's register tokens already absorb the global-context contamination that the CLS-subtract trick is designed to remove from CLIP's CLS — so subtracting again over-corrects. The λ=0.3 was tuned for OpenAI CLIP-B/16 in the SegEarth-OV paper.

## Test plan

- [x] All three bug fixes verified end-to-end through the running preview (datasets visible without upload; `avg_conf 0.152 → 0.739`; `total_area_m2 ~0 → 2.43 × 10⁹`; class % sums to 100)
- [x] Bench reproduces the published TIPSv2-L/14 25.06 mIoU within 0.07 (values + 9 templates)
- [x] Best new config (9 templates + M-SA) beats published number by +0.81 on the full 2000 val images
- [x] Ablation matrix (n=200) isolates the contribution of each individual change
- [x] No new console errors in preview (only pre-existing maplibre tile-fetch errors remain)
- [x] Existing `classify_image_zeroshot` / `auto_label_geotiff_tipsv2` API unchanged — bench imports the new helpers; production paths use defaults

## Notes for reviewers

- The bench downloads ~1 GB the first time it runs (`backend/data/ade20k/ADEChallengeData2016.zip`). `backend/data/` is gitignored.
- The `cls_subtract` and `drop_residual` knobs are wired through but **off by default** since on TIPSv2 they're at best neutral and at worst regressive on ADE20K. They're there for users running the bench against RS imagery (where SegEarth-OV's paper shows them helpful on CLIP-B/16; TIPSv2's behaviour may differ).
- `attn_mode="msa"` is the new best default for dense seg. The existing production path (`classify_image_zeroshot`) wasn't switched as part of this PR — it still uses the model's default forward — so user-facing behaviour is unchanged. A follow-up could expose `attn_mode` as a knob on `auto_label_geotiff_tipsv2`.

## References

- [TIPSv2 (CVPR'26)](https://github.com/google-deepmind/tips) — model + official zero-shot segmentation notebook
- [SegEarth-OV (CVPR'25)](https://arxiv.org/abs/2410.01768) — CLS subtraction (Eq. 9), modulated self-attention (Eq. 10), SimFeatUp upsampler
- [ClearCLIP (ECCV'24)](https://www.ecva.net/papers/eccv_2024/papers_ECCV/papers/06346.pdf) — three last-block modifications for dense seg
- [GEOBench-VLM (Mar 2025)](https://arxiv.org/abs/2411.19325) — geospatial VLM benchmark; confirms there is no RS-specific dense-seg model in their evaluation set (only generic GLaMM at 14.11 mIoU on RS referring segmentation)

### CODE REVIEW

**Comment by 2imi9** [comment]
Closed — wrong session, opened by mistake.
